raspberrypi/daughter_cards/colorsensors.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+from common.serialtalks import BYTE, FLOAT
+from daughter_cards.arduino import SecureArduino
+
+CSA_GET_RGB_OPCODE = 0X10
+CSA_GET_CUB_COLOR_ESTIMATE_OPCODE = 0X11
+CSA_SET_INTEGRATION_TIME_OPCODE = 0X12
+CSA_SET_GAIN_OPCODE = 0X13
+CSA_ENABLE_OPCODE = 0X14
+CSA_DISABLE_OPCODE = 0X15
+CSA_LEDS_ON_OPCODE = 0X16
+CSA_LEDS_OFF_OPCODE = 0X17
+CSA_SET_CUP_COLOR_ESTIMATE_SAMPLE_SIZE_OPCODE = 0X18
+
+class ColorSensorArray(SecureArduino):
+    INTEGRATIONTIME_2_4MS = 0
+    INTEGRATIONTIME_24MS = 1
+    INTEGRATIONTIME_50MS = 2
+    INTEGRATIONTIME_101MS = 3
+    INTEGRATIONTIME_154MS = 4
+    INTEGRATIONTIME_700MS = 5
+    GAIN_1X = 0
+    GAIN_4X = 1
+    GAIN_16X = 2
+    GAIN_60X = 3
+    CUP_COLOR_ESTIMATE_RED = 0
+    CUP_COLOR_ESTIMATE_GREEN = 1
+    CUP_COLOR_ESTIMATE_ERROR = 2
+
+    def __init__(self, parent, uuid='ColorSensors'):
+        SecureArduino.__init__(self, parent, uuid, dict())
```

Comment:
Comme tu utillise le SecureArduino, tu peux definir des valeurs de retour par defaut en cas d'erreur de com.
```suggestion
DEFAULT = {CSA_GET_RGB_OPCODE: Deserializer(FLOAT(255)  + FLOAT(255) + FLOAT(255) ),
               etc... }
               
    def __init__(self, parent, uuid='ColorSensors'):
        SecureArduino.__init__(self, parent, uuid, self._DEFAULT)
```